_posts/2017-05-05-libz-blitz.md

```diff
@@ -0,0 +1,356 @@
+---
+layout: post
+title: "The Rust Libz Blitz"
+author: Brian Anderson, David Tolnay, and Aaron Turon
+description: "Improving the quality and maturity of Rust's core ecosystem"
+---
+
+To help bring our [2017 vision for Rust] to fruition, the Rust subteams are
+launching initiatives targeted at specific roadmap goals. **This post covers the
+library team's major initiative: raising a solid core of the Rust crate
+ecosystem to a consistent level of completeness and quality.** This work is
+guided in part by what we learned in producing the Rust standard library, and
+will result in a "cookbook" chock full of ready-made examples for common tasks,
+drawing from a careful selection of Rust crates.
+
+[2017 vision for Rust]: https://blog.rust-lang.org/2017/02/06/roadmap.html
+
+## Batteries included?
+
+**You can't be productive in a language without libraries that are easy to find
+and use.** Of course, the easiest library to find is the standard library, and
+many languages have taken a "batteries included" approach that includes APIs for
+a broad set of tasks directly in the standard library. There's a lot of upside
+to doing so: it means there's just one place to look for common tasks, with APIs
+that (hopefully!) cohere across tasks, compatibility across the ecosystem, and a
+promise of library maintenance. What's not to like?
+
+There's a countervailing mindset which, in its harshest terms, says "the
+standard library is where code goes to die" (a complaint sometimes heard about
+batteries-included standard libraries). Because the standard library is coupled
+to the language itself, it is released at the same frequency as the language is,
+and making breaking changes means breaking the whole language. For these reasons
+and others, standard libraries tend to evolve conservatively.
+
+As usual in the Rust world, our goal is to have our cake and eat it too. **Can
+we provide something like a batteries-included experience, without ossifying
+that code in the standard library?** This is a particularly vital question for
+Rust because, as a community, we are still rapidly exploring and iterating on
+best practices for ownership-based API design.
+
+The key ingredient to our approach is [Cargo], the package manager that shipped
+with Rust 1.0 and has been improving ever since. Cargo provides a unified
+dependency and workflow story across the entire Rust ecosystem, and makes adding
+dependencies a painless and relatively low-risk proposition. The ability to
+share code with ease is a powerful change to the character of, and audience for,
+traditional systems programming tasks.
+
+[Cargo]: https://blog.rust-lang.org/2016/05/05/cargo-pillars.html
+
+Because of Cargo, we can "include batteries" without literally putting them into
+the standard library; pulling in other crates is nearly as easy as using the
+standard library (and will get even easier this year). But to capitalize on this
```

Comment:
Out of curiosity, is "will get even easier this year" referring to something in particular?